Upgrade OTel (#4232)
grafana/alloy: docs/sources/shared/reference/components/otelcol-queue-batch-block.md

description: Shared content, otelcol queue batch block
headless: true
---

The following arguments are supported:

| Name            | Type        | Description                                                                                                | Default      | Required |
| --------------- | ----------- | ---------------------------------------------------------------------------------------------------------- | ------------ | -------- |
| `flush_timeout` | `duration`  | Time after which a batch will be sent regardless of its size. Must be a non-zero value.                    |              | yes      |
| `min_size`      | `number`    | The minimum size of a batch.                                                                               |              | yes      |
| `max_size`      | `number`    | The maximum size of a batch, enables batch splitting.                                                      |              | yes      |
| `sizer`         | `string`    | How the queue and batching is measured. Overrides the sizer set at the `sending_queue` level for batching. |              | yes      |

> `max_size` should be greater than or equal to `min_size`.

Reviewer: ```suggestion
`max_size` must be greater than or equal to `min_size`.
```

"should" implies it could be less that `min_size` as well.

> The `sizer` argument could be set to:

Reviewer: ```suggestion
The `sizer` argument can be set to:
```

> * `items`: number of the smallest parts of each signal (spans, metric data points, log records).

Reviewer: ```suggestion
* `items`: The number of the smallest parts of each span, metric data point, or log record.
```

Are these the only telemetry signals? If these are just common examples of many, then instead it could be:

```
* `items`: The number of the smallest parts of each telemetry signal.
```
